### ISSUE TITLE
Error encoded, decode

### ISSUE THREAD

**Issue by xyzhieu** [valhalla/valhalla#4251]
When decode and encode the following test case i don't get the desired result.

//vCoords_Case  != vCoords_Test 
[0] = (107.3749770, 11.5833730)
[1] = (107.3739679, 11.5830050)
[2] = (107.3729499, 11.5825950)
[3] = (107.3727600, 11.5825169)

//vCoords_Case  == vCoords_Test 
[0] = (107.3727600, 11.5825169)
[1] = (107.3729499, 11.5825950)
[2] = (107.3739679, 11.5830050)
[3] = (107.3749770, 11.5833730)

std::string shapeText = valhalla::midgard::encode(vCoords,1e7);
auto vCoords_Test = valhalla::midgard::decode<std::vector<std::pair<double, double>>>(res.shape,1e-7);

**Comment by kevinkreiser** [comment]
they are the same but reverse order?

**Comment by xyzhieu** [comment]
yes, I found out when encode with 7 digits, the above case overflows

**Comment by kevinkreiser** [comment]
yeah because we use 32bits internally per coordinate. this is not a bug really it wasn't meant to support arbitrary precision it was really for switching between 5 and 6 digits

**Comment by kevinkreiser** [comment]
I suppose if there is no performance penalty we could use 64bit integers. we'll have to test it

**Comment by nilsnolde** [comment]
We could also test vendoring this one: https://github.com/vahancho/polylineencoder.

**Comment by kevinkreiser** [comment]
i would vote against vendoring:

1. we already have an annoying amount of dependencies to manage, i prefer the devil i know :smile: 
2. our varint implementation supports using up to 7 bits of the byte, it would be stupid for us to rip out the canonical 5bit one for the sake of saving 100 lines of code when we'd have to keep the 7bit version for the sake of smaller tiles and backwards compatibility in the data
